### PR TITLE
Fix some items missing activities migrations.

### PR DESCRIPTION
**Pull request by Fyorl**

### CODE REVIEW

File: module/data/item/templates/activities.mjs [hunk 1]
```diff
@@ -231,8 +234,8 @@ export default class ActivitiesTemplate extends SystemDataModel {
     if ( !source.system.actionType && !source.system.activation?.type
       && (source.type !== "tool") ) return false;
 
-    // If item was updated after `4.0.0`, it shouldn't need the migration
-    if ( !foundry.utils.isNewerVersion("4.0.0", source._stats?.systemVersion ?? "0.0.0") ) return false;
+    // If item was updated after `4.0.1`, it shouldn't need the migration
+    if ( !foundry.utils.isNewerVersion("4.0.1", source._stats?.systemVersion ?? "0.0.0") ) return false;
```
Comment (Fyorl) [inline module/data/item/templates/activities.mjs:238]: We don't want to run the risk of re-migrating items that have already migrated to activities, but it feels reasonably safe to bump this version check in order to catch the items that were missed with the original migration. Any item that already has activities won't get hit a second time, and it's probably too early for many people to be doing things like deleting activities off their items that auto-migrated in 4.0.

File: module/data/item/templates/activities.mjs [hunk 2]
```diff
@@ -185,9 +185,12 @@ export default class ActivitiesTemplate extends SystemDataModel {
       }
     }
 
-    // If period is not blank, set recovery type to `recoverAll`
+    // If period is not blank, set an appropriate recovery type
     else if ( source.uses.per ) {
-      source.uses.recovery = [{ period: source.uses.per, type: "recoverAll" }];
+      if ( CONFIG.DND5E.limitedUsePeriods[source.uses.per]?.formula && source.uses.recovery ) {
```
Comment (Fyorl) [inline module/data/item/templates/activities.mjs:190]: This fixes an issue where `dawn` and `dusk` weren't migrated to use a formula even though their configuration has `formula: true`.

File: module/documents/item.mjs [hunk 3]
```diff
@@ -1846,8 +1846,8 @@ export default class Item5e extends SystemDocumentMixin(Item) {
 
   /** @inheritDoc */
   static migrateData(source) {
-    ActivitiesTemplate.initializeActivities(source);
     source = super.migrateData(source);
+    ActivitiesTemplate.initializeActivities(source);
```
[the remark this answers is not in the document]
Comment (arbron) [reply module/documents/item.mjs:1850]: I think it was originally moved to ensure uses migration worked properly when there were charges & other item uses, but I gave it a test and it seems to work fine after `super.migrateData`.

**Review by arbron — APPROVED** [review]

File: module/documents/item.mjs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Fyorl) [reply module/documents/item.mjs:1850]: I think I see the issue. For recharge features, you end up with a recharge on the item itself, as well as a recharge on the auto-created activity.

*Migration behaviour before*
image

*Migration behaviour after*
image

I think I can probably fix this by tweaking `transformUsesData`.

**Review by arbron — APPROVED** [review]
Looking good

**Pull request merged**